addons/web/static/src/core/popover/popover_hook.js

```diff
@@ -51,7 +59,63 @@ export function usePopover(component, options = {}) {
             options.onClose?.();
         }
     };
-    const popover = makePopover(popoverService, component, newOptions);
+    const popover = makePopover(popoverService.add, component, newOptions);
     onWillUnmount(popover.close);
     return popover;
 }
+
+class PopoverAsDialog extends Component {
+    static components = { Dialog };
+    static props = ["close", "component", "componentProps"];
+    static template = xml`
+        <Dialog footer="false">
+            <t t-component="props.component" t-props="componentProps"/>
+        </Dialog>
+    `;
+    get componentProps() {
+        return { ...this.props.componentProps, close: this.props.close };
+    }
+}
+
+/**
+ * Manages a component to be used as a popover.
+ * Replaced by a fullscreen dialog on small screens.
+ *
+ * @param {typeof import("@odoo/owl").Component} component
+ * @param {import("@web/core/popover/popover_service").PopoverServiceAddOptions} [options]
+ * @returns {PopoverHookReturnType}
+ */
+export function useResponsivePopover(component, options = {}) {
+    const popover = usePopover(...arguments);
+
+    const dialogService = useService("dialog");
+    const owner = useComponent();
+    const newOptions = Object.create(options);
+    newOptions.onClose = () => {
+        if (status(owner) !== "destroyed") {
+            options.onClose?.();
+        }
+    };
+    const popoverAsDialog = makePopover(
+        (_, comp, props, options) => {
+            const wrapperProps = { component, componentProps: props };
+            return dialogService.add(comp, wrapperProps, options);
+        },
+        PopoverAsDialog,
+        pick(newOptions, "onClose")
+    );
+    onWillUnmount(popoverAsDialog.close);
+
+    const getResponsive = () => (owner.env.isSmall ? popoverAsDialog : popover);
+    return {
```

Comment:
is it possible to open a PopoverAsDialog and then close a Popover because env.isSmall change?